packages/frontend/src/components/profile/Profile.js

```diff
@@ -164,6 +165,13 @@ export function Profile({ match }) {
     const userRecoveryMethods = useSelector((state) => selectRecoveryMethodsByAccountId(state, { accountId: account.accountId }));
     const twoFactor = has2fa && userRecoveryMethods && userRecoveryMethods.filter((m) => m.kind.includes('2fa'))[0];
 
+    const accountState = useSelector(selectAccountSlice);
+    const keys = accountState.fullAccessKeys || [];
+    const publicKeys = keys.map((key) => key.public_key);
+    const hasLedger = userRecoveryMethods.filter((method) => method.kind === 'ledger').map((key) => key.publicKey).some((key) => publicKeys.includes(key));
```

Comment:
Haven't we already filtered out recovery methods that do not have a key on chain in [`getRecoveryMethods`](https://github.com/near/near-wallet/blob/master/packages/frontend/src/utils/wallet.js#L912)? Seems any selector should only have recovery methods with keys confirmed on chain.